### PR TITLE
feat(FN-1456): quarterly reports on TFM bank reports page

### PR DESCRIPTION
**Pull request by RLavender98**
## Introduction :pencil2:
Include quarterly reports on the TFM reports page

## Resolution :heavy_check_mark:
- FN-1456: Current reporting period section contains current quarterly reports and if there are any include the quarterly report period in the heading
- FN-1412: Open reports sections contain open quarterly reports, grouped by their submission month with monthly reports, and the header details the quarterly report period if there is one

## Miscellaneous :heavy_plus_sign:
- Also closes off FN-1799, using shortened months for report period headers, this was pre-existing behaviour for the monthly report periods included in headers and has been continued to be used for the quarterly report periods in headers.

### CODE REVIEW

File: libs/common/src/utils/report-period.ts [hunk 1]
```diff
@@ -151,19 +126,32 @@ export const getCurrentReportPeriodForBankSchedule = (bankReportPeriodSchedule:
...
+
+  const formattedEndOfPeriod = format(endOfReportPeriod, `${monthFormat} yyyy`);
+  if (isEqualMonthAndYear(reportPeriod.start, reportPeriod.end)) {
+    return includePeriodicity ? `${formattedEndOfPeriod} (monthly)` : formattedEndOfPeriod;
+  }
+
+  const formattedStartOfPeriod =
+    reportPeriod.start.year === reportPeriod.end.year ? format(startOfReportPeriod, monthFormat) : format(startOfReportPeriod, `${monthFormat} yyyy`);
+  return includePeriodicity ? `${formattedStartOfPeriod} to ${formattedEndOfPeriod} (quarterly)` : `${formattedStartOfPeriod} to ${formattedEndOfPeriod}`;
 };
```
Comment (francescastocco) [inline libs/common/src/utils/report-period.ts:151]: I like this!

File: dtfs-central-api/src/v1/controllers/utilisation-report-service/get-utilisation-reports-reconciliation-summary.controller/helpers.test.ts
Comment (francescastocco) [inline dtfs-central-api/src/v1/controllers/utilisation-report-service/get-utilisation-reports-reconciliation-summary.controller/helpers.test.ts:1]: These are really comprehensive 👏

**Review by francescastocco — COMMENTED** [review]
Looks great, just one comment but otherwise happy to approve

File: trade-finance-manager-ui/server/controllers/utilisation-reports/helpers/reconciliation-summary-helper.ts [hunk 2]
```diff
@@ -67,12 +79,13 @@ export const getReportReconciliationSummariesViewModel = async (
   return summariesApiResponse.map(({ items: apiItems, submissionMonth }) => {
     const reportDueDate = getReportDueDate(bankHolidays, submissionMonth);
     const reportPeriodStart = getReportPeriodStart(submissionMonth);
+    const distinctReportPeriods = getDistinctReportPeriodsByStartMonth(apiItems.map((item) => item.reportPeriod));
 
     return {
       items: apiItems.map(getSummaryItemViewModel),
       submissionMonth,
       reportPeriodStart,
```
[the remark this answers is not in the document]
Comment (RLavender98) [reply trade-finance-manager-ui/server/controllers/utilisation-reports/helpers/reconciliation-summary-helper.ts:87]: Agh this does seem like something we definitely don't want to use...

File: dtfs-central-api/src/v1/controllers/utilisation-report-service/get-utilisation-reports-reconciliation-summary.controller/helpers.test.ts
Comment (RLavender98) [reply dtfs-central-api/src/v1/controllers/utilisation-report-service/get-utilisation-reports-reconciliation-summary.controller/helpers.test.ts:1]: 😊

**Review by francescastocco — APPROVED** [review]
LGTM & tests are all passing :)

**Pull request merged**